testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/container/BaseContainer.java

```diff
@@ -0,0 +1,72 @@
+package org.wildfly.test.integration.observability.container;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.WaitAllStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+public abstract class BaseContainer<SELF extends GenericContainer<SELF>> extends GenericContainer<SELF> {
+    private final String containerName;
+    private final List<Integer> exposedPorts;
+
+    private static final int STARTUP_ATTEMPTS = Integer.parseInt(
+            System.getProperty("testsuite.integration.container.startup.attempts", "1"));
+    private static final Duration ATTEMPT_DURATION = Duration.parse(
+            System.getProperty("testsuite.integration.container.attempt.duration", "PT5S"));
+    protected final Boolean enableLogging = true; // Default: null/false
+
+    public BaseContainer(String containerName,
+                         String imageName,
+                         String imageVersion,
+                         List<Integer> exposedPorts,
+                         List<WaitStrategy> waitStrategies) {
+        super(DockerImageName.parse(imageName + ":" + imageVersion));
+
+        this.containerName = containerName;
+        this.exposedPorts = exposedPorts;
+
+        setWaitStrategy(buildWaitStrategy(waitStrategies));
+        setExposedPorts(exposedPorts);
+        setStartupAttempts(STARTUP_ATTEMPTS);
+
+//        enableLogging =
+//                Boolean.parseBoolean(System.getProperty("testsuite.integration.container.logging")) ||
+//                Boolean.parseBoolean(System.getProperty("testsuite.integration.container." + containerName.toLowerCase() + ".logging"));
+
+        if (enableLogging) {
+            setLogConsumers(List.of(outputFrame -> {
+                byte[] bytes = outputFrame.getBytes();
+                if (bytes != null) {
+                    debugLog(new String(bytes));
+                }
+            }));
+        }
+    }
+
+    @Override
+    public SELF withExposedPorts(Integer... ports) {
+        getExposedPorts().addAll(exposedPorts);
```

Comment:
@jasondlee I don't see what this does.

GenericContainer.getExposedPorts() returns a list that is only referred to in a local variable. So modifying that list and then letting that value go out of scope doesn't affect anything.

Also if the intent is to ignore the 'ports' param and just rely on the 'exposedPorts' passed into the constructor, the javadoc should say that.

Reply:
That's an override from the super class that doesn't appear to be needed anymore, so I'll remove it.